### PR TITLE
Convert course settings page to async/await

### PR DESCRIPTION
**Pull request by mylesw27**
This PR intends to convert callback functions to `async/await` in the instructorCourseAdminSettingsPage. This is part of the larger modernization efforts and future PRs will convert this page to TypeScript.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.js [hunk 1]
```diff
@@ -1,53 +1,41 @@
...
+  asyncHandler(async (req, res) => {
+    try {
+      await fs.access(res.locals.course.path);
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        res.locals.needToSync = true;
+      } else {
+        throw new Error(err);
+      }
+    }
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.js:24]: Now that I see this written out, I think using [`pathExists` from `fs-extra`](https://github.com/jprichardson/node-fs-extra/blob/master/docs/pathExists.md) is probably the cleaner solution. I asked for opinions about this on Slack; we'll see if anyone feels strongly either way. Hold off on changing this for right now!

Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.js:24]: As discussed on Slack, we'll go with `pathExists` here.

File: apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.js [hunk 2]
```diff
@@ -1,53 +1,32 @@
...
-  debug('GET /');
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const pathExists = await fs.pathExists(res.locals.course.path);
+    if (!pathExists) {
+      res.locals.needToSync = true;
+    } else {
+      const jsonExists = await fs.pathExists(path.join(res.locals.course.path, 'infoCourse.json'));
+      console.log('jsonExists', jsonExists);
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.js:21]: Extraneous `console.log`.

Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.js]: Removed in 477c308.

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**